src/uct/ib/mlx5/rc/rc_mlx5_iface.c

```diff
@@ -421,7 +418,9 @@ static ucs_status_t uct_rc_mlx5_iface_preinit(uct_rc_mlx5_iface_common_t *iface,
     ucs_status_t status;
 
     status = uct_rc_mlx5_iface_parse_srq_topo(md, mlx5_config,
-                                              &iface->config.srq_topo);
+                                              &iface->config.srq_topo,
+                                              !!(init_attr->flags &
+                                                 UCT_IB_DDP_SUPPORTED));
```

Comment:
what if DDP is supported, but disabled by configuration? would we also fallback to cyclic_emulated (we should not)?